src/test/java/de/tum/in/www1/artemis/ArchitectureTest.java

```diff
@@ -107,6 +107,22 @@ void testLogging() {
         GeneralCodingRules.NO_CLASSES_SHOULD_ACCESS_STANDARD_STREAMS.check(classes);
     }
 
+    @Test
+    void testJSONImplementations() {
+        ArchRule jsonObject = noClasses().should()
+                .dependOnClassesThat(have(simpleName("JsonObject")).or(have(simpleName("JSONObject"))).and(not(resideInAPackage("com.google.gson"))));
```

Comment:
I think that makes it a bit more readable ("have name x or name y" instead of "have name x or have name y")
```suggestion
                .dependOnClassesThat(have(simpleName("JsonObject").or(simpleName("JSONObject"))).and(not(resideInAPackage("com.google.gson"))));
```

Reply:
You are right, I implemented your suggestion.